### ISSUE TITLE
Dispatch refresh / create normalizing properties doesn't work

### ISSUE THREAD

**Issue by edemehmetcan** [microsoft/botbuilder-tools#1477]
What project is affected?
Dispatch refresh/create command

What language is this in?

Powershell

What happens?
When adding the properties in the dispatcher file:
"normalizePunctuation": true/false
"normalizeDiacritics": true/false

Commands are not working after launching the command. After checking on the LUIS portal, I don't see that the option is turn on/off. Please take a look on this. 

What are the steps to reproduce this issue?

Create a dispatcher file
put option : "normalizePunctuation": true (for example)
and do a refresh

What were you expecting to happen?
Options working in LUIS app. 

Can you share any logs, error output, etc.?
Any screenshots or additional context?

**Comment by tsuwandy** [comment]
@Zerryth, could you try to repro?  Thanks!

**Comment by Zerryth** [comment]
@edemehmetcan did you find documentation indicating that Dispatch accepts flags `normalizePunctuation` and `normalizeDiacritic`? This [Dispatch README](https://github.com/microsoft/botbuilder-tools/tree/master/packages/Dispatch) doesn't seem to indicate that it does accept those commands currently 🤔.

___

The documentation that I could find around this flag were specific to the LUIS API, which you could configure in the [LUIS portal under Manage](https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-application-settings).

Or, if you're looking for a programmatic solution, using the API for version update:
https://westus.dev.cognitive.microsoft.com/docs/services/5890b47c39e2bb17b84a55ff/operations/versions-update-application-version-settings/console

The API even has fields that you could fill in to test with your live app. For example, filling in `appId`, `versionId`, and `Ocp-Apim-Subscription-Key` (LUIS authoring key), then changing `NormalizePunctuation` to `true`:

![image](https://user-images.githubusercontent.com/35248895/97502686-dd7ccb80-1930-11eb-8606-745da6280837.png)

PUT https://westus.api.cognitive.microsoft.com/luis/api/v2.0/apps/17f7xxxx-xxxx-xxxx-xxxx-xxxxxxxxa043/versions/0.1/settings HTTP/1.1
Host: westus.api.cognitive.microsoft.com
Content-Type: application/json
Ocp-Apim-Subscription-Key: ••••••••••••••••••••••••••••••••

[
	{
		"name": "UseAllTrainingData",
		"value": "true"
	},
	{
		"name": "NormalizeDiacritics",
		"value": "false"
	},
	{
		"name": "NormalizePunctuation",
		"value": "true"
	}
]

If the request is successful changes are visible in the LUIS portal under Manage > Settings

![image](https://user-images.githubusercontent.com/35248895/97502847-2fbdec80-1931-11eb-8b2d-84ec1a6de742.png)

**Comment by edemehmetcan** [comment]
> @edemehmetcan did you find documentation indicating that Dispatch accepts flags `normalizePunctuation` and `normalizeDiacritic`? This [Dispatch README](https://github.com/microsoft/botbuilder-tools/tree/master/packages/Dispatch) doesn't seem to indicate that it does accept those commands currently 🤔.
> 
> The documentation that I could find around this flag were specific to the LUIS API, which you could configure in the [LUIS portal under Manage](https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-application-settings).
> 
> Or, if you're looking for a programmatic solution, using the API for version update:
> https://westus.dev.cognitive.microsoft.com/docs/services/5890b47c39e2bb17b84a55ff/operations/versions-update-application-version-settings/console
> 
> The API even has fields that you could fill in to test with your live app. For example, filling in `appId`, `versionId`, and `Ocp-Apim-Subscription-Key` (LUIS authoring key), then changing `NormalizePunctuation` to `true`:
> 
> ![image](https://user-images.githubusercontent.com/35248895/97502686-dd7ccb80-1930-11eb-8606-745da6280837.png)
> 
> PUT https://westus.api.cognitive.microsoft.com/luis/api/v2.0/apps/17f7xxxx-xxxx-xxxx-xxxx-xxxxxxxxa043/versions/0.1/settings HTTP/1.1
> Host: westus.api.cognitive.microsoft.com
> Content-Type: application/json
> Ocp-Apim-Subscription-Key: ••••••••••••••••••••••••••••••••
> 
> [
> {
> "name": "UseAllTrainingData",
> "value": "true"
> },
> {
> "name": "NormalizeDiacritics",
> "value": "false"
> },
> {
> "name": "NormalizePunctuation",
> "value": "true"
> }
> ]
> 
> If the request is successful changes are visible in the LUIS portal under Manage > Settings
> 
> ![image](https://user-images.githubusercontent.com/35248895/97502847-2fbdec80-1931-11eb-8b2d-84ec1a6de742.png)

Dear

After creating the dispatch file, it’s automatically putting this line in the .dispatch file : 

"normalizeDiacritics": true

I didn’t found the documentation as well indicating that the dispatcher can have that type of configuration but logically it should be possible to activate or not these configuration ? Dispatcher is a LUIS app so why I can’t put it in the config ? After each refresh I need to turn on normalize punctuation manually, weird :/

**Comment by Zerryth** [comment]
> After creating the dispatch file, it’s automatically putting this line in the .dispatch file :

@edemehmetcan could you please specify what commands you're running to create the dispatch file?
For example, `dispatch init...`, dispatch add...`. Want to know what kinds of apps you're putting into your dispatch app, so I could repro and understand the issue better. (For example, are you adding child LUIS models to your app? Or QnA Maker apps? etc.)

>  didn’t found the documentation as well indicating that the dispatcher can have that type of configuration but logically it should be possible to activate or not these configuration ? Dispatcher is a LUIS app so why I can’t put it in the config ?

I suppose there's the misconception there. Dispatcher is technically a tool that does try to create a giant, merged LUIS model to route you to appropriate sub models, whether they be child LUIS models or QnA Maker, etc., however not every single LUIS API is available in the Dispatcher tool.

I suspect that's why when you try passing in normalizeDiacritics/normalizePunctuation into Dispatch, it does nothing to the parent LUIS model that Dispatch creates--it seems like it's a LUIS API that isn't available in Dispatch. 

___

Could you please elaborate more on the exact steps/commands that you're running? Want to understand the scenario, why you are having to `normalize punctuation manually` after each refresh

Then the engineers who specialize on Dispatch could either prioritize this into a feature request or perhaps we could suggest a workaround where you're not having to update LUIS version settings after every refresh

**Comment by edemehmetcan** [comment]
@Zerryth I am creating the dispatch file firsly by init and add (LUIS and QnA apps).

In order to focus to the problem, I have a bunch of question in QnA and when adding them into the dispatch, when I ask a question with and without the question mark I got different score. The question marks from KBs are deleted by default after adding to dispatcher. 

So when I ask a question:

-  when adding the (?) in the question the dispatcher show me a prediction score of 0.5-0.6 and 
- when asking the question without the (?) I got a score of 0.8-0.9

After enabling the "normalizePunctuation" manually this issue is gone. I got the same prediction score...

**Comment by Zerryth** [comment]
@edemehmetcan, so I did some poking around.

### The Workaround

It looks like you can make dispatch create a LUIS model with your desired values for `normalizeDiacritics` and `normalizePunctuation`. There isn't a built-in command-line commandlet, however just edit your `.dispatch` file directly to have the desired values.

For example, here I set both `normalizeDiacritics` and `normalizePunctuation` to `true` in my `.dispatch` file:

```json
{
  "authoringRegion": "westus",
  "culture": "es-es",
  "hierarchical": true,
  "useAllTrainingData": true,
  "dontReviseUtterance": false,
  "copyLuisData": true,
  "normalizeDiacritics": true,
  "normalizePunctuation": true,
  "services": [
    {
      "includePrompts": true,
      "kbId": "7b13xxxx-xxxx-xxxx-xxxx-xxxxxxxx511e",
      "subscriptionKey": "94c5axxxxxxxxxxxxxxxxxxxxxxxxa27",
      "type": "qna",
      "name": "Faq"
    }
  ],
  "serviceIds": [
    "1"
  ],
  "appId": "918xxxx-xxxx-xxxx-xxxx-xxxxxxxx977a",
  "authoringKey": "63e7xxxxxxxxxxxxxxxxxxxxxxxx37a8",
  "version": "Dispatch",
  "region": "westus",
  "type": "dispatch",
  "name": "normalization"
}
```

Then when I `dispatch create` or `dispatch refresh`, it will carry the values in the dispatch file over to the parent LUIS model that it creates/updates, which you can verify in the LUIS portal under Manage > Settings > Version Settings.

The training data values from QnA Maker that I added to Dispatch still show up in LUIS with diacritics and punctuation, however when you actually query, you notice that the prediction scores are unaffected by punctuation and diacritics if both normalization values are set to `true`.
___

### Testing

Utterance that I have in the LUIS model created by dispatch: `volveré...` 

![image](https://user-images.githubusercontent.com/35248895/97617174-b4187a00-19da-11eb-89a6-b6a6c9f1e315.png)

For 1. I have both normalization values set to `true`, and you can see that the score is the same for all 3 utterances where I test combinations with and without diacritics and punctuation. 

Then for 2. in the LUIS portal I toggled the normalize diacritics and punctuation to off (which will turn it to `false` in LUIS's backend). Trained to save the setting in LUIS. Then testing again with and without diacritics and punctuation, you see I receive different scores now that normalization is off.

___

Closing this issue as the team isn't currently adding new features to Dispatch, since our efforts are focused on fine-tuning Orchestrator, the new, shiny, Dispatch CLI, which is in preview currently:
https://www.npmjs.com/package/@microsoft/bf-orchestrator-cli
https://www.nuget.org/packages/Microsoft.BotFramework.Orchestrator.Managed/
